[2025-03-20 10:05:24,348: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:05:24,351: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:05:24,352: INFO: common: created directory at: artifacts]
[2025-03-20 10:05:24,352: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:05:24,353: INFO: 1386379862: Loading generator and discriminator models]
[2025-03-20 10:05:24,821: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:05:24,823: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:05:24,824: INFO: 1386379862: Preparing dataset]
[2025-03-20 10:08:35,019: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:08:35,022: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:08:35,022: INFO: common: created directory at: artifacts]
[2025-03-20 10:08:35,023: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:08:35,023: INFO: 2522334285: Loading generator and discriminator models]
[2025-03-20 10:08:35,448: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:08:35,450: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:08:35,452: INFO: 2522334285: Preparing dataset]
[2025-03-20 10:12:33,844: ERROR: base_events: Task was destroyed but it is pending!
task: <Task pending name='Task-3' coro=<Kernel.poll_control_queue() running at /Applications/AI/GAN/.venv/lib/python3.9/site-packages/ipykernel/kernelbase.py:309> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x10535fd30>()]> cb=[_chain_future.<locals>._call_set_state() at /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/asyncio/futures.py:391]>]
[2025-03-20 10:12:45,797: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:12:45,800: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:12:45,801: INFO: common: created directory at: artifacts]
[2025-03-20 10:12:45,801: INFO: 3821775565: Found training data at: artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:12:45,802: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:12:45,802: INFO: 3821775565: Loading generator and discriminator models]
[2025-03-20 10:12:46,250: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:12:46,252: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:12:46,254: INFO: 3821775565: Preparing dataset]
[2025-03-20 10:12:46,280: INFO: 3821775565: Directory structure not found, trying direct image loading]
[2025-03-20 10:12:46,282: INFO: 3821775565: Found 300 image files]
[2025-03-20 10:12:46,361: INFO: 3821775565: Starting GAN training for 100 epochs]
[2025-03-20 10:12:46,361: INFO: 3821775565: Epoch 1/100]
[2025-03-20 10:13:44,950: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:13:44,953: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:13:44,954: INFO: common: created directory at: artifacts]
[2025-03-20 10:13:44,954: INFO: 243411336: Found training data at: artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:13:44,955: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:13:44,955: INFO: 243411336: Loading generator and discriminator models]
[2025-03-20 10:13:45,397: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:13:45,399: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:13:45,401: INFO: 243411336: Preparing dataset]
[2025-03-20 10:13:45,423: INFO: 243411336: Directory structure not found, trying direct image loading]
[2025-03-20 10:13:45,424: INFO: 243411336: Found 300 image files]
[2025-03-20 10:13:45,498: INFO: 243411336: Starting GAN training for 100 epochs]
[2025-03-20 10:13:45,499: INFO: 243411336: Epoch 1/100]
[2025-03-20 10:15:41,495: ERROR: base_events: Task was destroyed but it is pending!
task: <Task pending name='Task-2' coro=<Kernel.poll_control_queue() running at /Applications/AI/GAN/.venv/lib/python3.9/site-packages/ipykernel/kernelbase.py:309> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x105717610>()]> cb=[_chain_future.<locals>._call_set_state() at /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/asyncio/futures.py:391]>]
[2025-03-20 10:15:52,332: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:15:52,335: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:15:52,336: INFO: common: created directory at: artifacts]
[2025-03-20 10:15:52,336: INFO: 527437301: Found training data at: artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:15:52,337: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:15:52,337: INFO: 527437301: Loading generator and discriminator models]
[2025-03-20 10:15:52,775: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:15:52,776: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:15:52,778: INFO: 527437301: Preparing datasets]
[2025-03-20 10:15:52,808: INFO: 527437301: Found Monet images in artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:15:52,809: INFO: 527437301: Found photo images in artifacts/data_ingestion/gan-getting-started/photo_jpg]
[2025-03-20 10:15:52,934: INFO: 527437301: Starting GAN training for 100 epochs]
[2025-03-20 10:17:12,806: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:17:12,808: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:17:12,809: INFO: common: created directory at: artifacts]
[2025-03-20 10:17:12,810: INFO: 2713986983: Found training data at: artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:17:12,810: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:17:12,811: INFO: 2713986983: Loading generator and discriminator models]
[2025-03-20 10:17:13,273: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:17:13,274: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:17:13,276: INFO: 2713986983: Preparing datasets]
[2025-03-20 10:17:13,308: INFO: 2713986983: Found Monet images in artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:17:13,309: INFO: 2713986983: Found photo images in artifacts/data_ingestion/gan-getting-started/photo_jpg]
[2025-03-20 10:17:13,442: INFO: 2713986983: Starting GAN training for 100 epochs]
[2025-03-20 10:17:13,479: INFO: 2713986983: Epoch 1/100]
[2025-03-20 10:18:22,680: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:18:22,682: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:18:22,683: INFO: common: created directory at: artifacts]
[2025-03-20 10:18:22,683: INFO: 2118690621: Found training data at: artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:18:22,684: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:18:22,684: INFO: 2118690621: Loading generator and discriminator models]
[2025-03-20 10:18:23,100: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:18:23,101: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:18:23,102: INFO: 2118690621: Preparing datasets]
[2025-03-20 10:18:23,133: INFO: 2118690621: Found Monet images in artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:18:23,134: INFO: 2118690621: Found photo images in artifacts/data_ingestion/gan-getting-started/photo_jpg]
[2025-03-20 10:18:23,252: INFO: 2118690621: Starting GAN training for 100 epochs]
[2025-03-20 10:18:23,288: INFO: 2118690621: Epoch 1/100]
[2025-03-20 10:18:36,942: INFO: 2118690621: Step 0/100 - Gen loss: 0.6571, Disc loss: 0.7906]
[2025-03-20 10:20:12,271: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:20:12,274: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:20:12,275: INFO: common: created directory at: artifacts]
[2025-03-20 10:20:12,276: INFO: 2118690621: Found training data at: artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:20:12,277: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:20:12,277: INFO: 2118690621: Loading generator and discriminator models]
[2025-03-20 10:20:12,720: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:20:12,722: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:20:12,723: INFO: 2118690621: Preparing datasets]
[2025-03-20 10:20:12,755: INFO: 2118690621: Found Monet images in artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:20:12,755: INFO: 2118690621: Found photo images in artifacts/data_ingestion/gan-getting-started/photo_jpg]
[2025-03-20 10:20:12,876: INFO: 2118690621: Starting GAN training for 1 epochs]
[2025-03-20 10:20:12,913: INFO: 2118690621: Epoch 1/1]
[2025-03-20 10:20:27,540: INFO: 2118690621: Step 0/100 - Gen loss: 0.6578, Disc loss: 0.7901]
[2025-03-20 10:22:41,992: INFO: 2118690621: Step 10/100 - Gen loss: 0.8416, Disc loss: 0.6729]
[2025-03-20 10:25:34,732: INFO: 2118690621: Step 20/100 - Gen loss: 0.7542, Disc loss: 0.7589]
[2025-03-20 10:27:55,903: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-03-20 10:27:55,909: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:27:55,914: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:27:55,914: INFO: common: created directory at: artifacts]
[2025-03-20 10:27:55,914: INFO: common: created directory at: artifacts/data_ingestion]
[2025-03-20 10:27:55,915: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/18Xl8070XZ4iEHhzalwdOmzZEH6Cmokh5/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2025-03-20 10:28:19,728: INFO: 2118690621: Step 30/100 - Gen loss: 0.7239, Disc loss: 0.6922]
[2025-03-20 10:29:55,477: INFO: data_ingestion: Downloaded data from https://drive.google.com/file/d/18Xl8070XZ4iEHhzalwdOmzZEH6Cmokh5/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2025-03-20 10:30:07,232: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-03-20 10:30:07,233: INFO: main: >>>>>> stage Prepare GAN Model stage started <<<<<<]
[2025-03-20 10:30:07,242: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:30:07,247: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:30:07,248: INFO: common: created directory at: artifacts]
[2025-03-20 10:30:07,249: INFO: common: created directory at: artifacts/prepare_gan_model]
[2025-03-20 10:30:07,249: INFO: prepare_gan_model: Preparing GAN models]
[2025-03-20 10:30:07,249: INFO: prepare_gan_model: Building generator model]
[2025-03-20 10:30:08,253: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:30:08,259: WARNING: __init__: There is a known slowdown when using v2.11+ Keras optimizers on M1/M2 Macs. Falling back to the legacy Keras optimizer, i.e., `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:30:08,277: INFO: prepare_gan_model: Building discriminator model]
[2025-03-20 10:30:08,596: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:30:08,597: WARNING: __init__: There is a known slowdown when using v2.11+ Keras optimizers on M1/M2 Macs. Falling back to the legacy Keras optimizer, i.e., `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:30:09,185: INFO: prepare_gan_model: Generator model saved at: artifacts/prepare_gan_model/generator_model.h5]
[2025-03-20 10:30:09,186: INFO: prepare_gan_model: Discriminator model saved at: artifacts/prepare_gan_model/discriminator_model.h5]
[2025-03-20 10:30:09,187: INFO: main: >>>>>> stage Prepare GAN Model stage completed <<<<<<

x==========x]
[2025-03-20 10:30:09,187: INFO: main: >>>>>> stage GAN Training stage started <<<<<<]
[2025-03-20 10:30:09,193: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 10:30:09,198: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 10:30:09,199: INFO: common: created directory at: artifacts]
[2025-03-20 10:30:09,213: INFO: common: created directory at: artifacts/gan_training]
[2025-03-20 10:30:09,215: INFO: train_gan: Loading generator and discriminator models]
[2025-03-20 10:30:10,998: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:30:11,000: WARNING: optimizer: At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.]
[2025-03-20 10:30:11,001: INFO: train_gan: Preparing datasets]
[2025-03-20 10:30:11,082: INFO: train_gan: Found Monet images in artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 10:30:11,083: INFO: train_gan: Found photo images in artifacts/data_ingestion/gan-getting-started/photo_jpg]
[2025-03-20 10:30:11,670: INFO: train_gan: Starting GAN training for 1 epochs]
[2025-03-20 10:30:11,920: INFO: train_gan: Epoch 1/1]
[2025-03-20 10:30:30,485: INFO: 2118690621: Step 40/100 - Gen loss: 0.7484, Disc loss: 0.6659]
[2025-03-20 10:30:47,198: INFO: train_gan: Step 0/100 - Gen loss: 0.6581, Disc loss: 0.7770]
[2025-03-20 10:35:09,666: INFO: 2118690621: Step 50/100 - Gen loss: 0.7251, Disc loss: 0.7092]
[2025-03-20 10:35:29,622: INFO: train_gan: Step 10/100 - Gen loss: 0.8715, Disc loss: 0.6340]
[2025-03-20 10:38:33,510: INFO: 2118690621: Step 60/100 - Gen loss: 0.7809, Disc loss: 0.6948]
[2025-03-20 10:38:56,868: INFO: train_gan: Step 20/100 - Gen loss: 0.8314, Disc loss: 0.6849]
[2025-03-20 10:42:13,028: INFO: 2118690621: Step 70/100 - Gen loss: 0.7402, Disc loss: 0.7028]
[2025-03-20 10:42:34,414: INFO: train_gan: Step 30/100 - Gen loss: 0.7572, Disc loss: 0.6750]
[2025-03-20 10:45:36,306: INFO: 2118690621: Step 80/100 - Gen loss: 0.7339, Disc loss: 0.6824]
[2025-03-20 10:45:56,958: INFO: train_gan: Step 40/100 - Gen loss: 0.7142, Disc loss: 0.7162]
[2025-03-20 10:49:23,004: INFO: 2118690621: Step 90/100 - Gen loss: 0.6941, Disc loss: 0.6913]
[2025-03-20 10:49:48,163: INFO: train_gan: Step 50/100 - Gen loss: 0.7158, Disc loss: 0.7006]
[2025-03-20 10:54:07,747: INFO: 2118690621: Time for epoch 1: 2034.83 sec]
[2025-03-20 10:54:08,847: INFO: 2118690621: Final models saved at artifacts/gan_training/generator_final.h5 and artifacts/gan_training/discriminator_final.h5]
[2025-03-20 10:54:08,849: INFO: 2118690621: GAN training completed]
[2025-03-20 10:54:38,474: INFO: train_gan: Step 60/100 - Gen loss: 0.7073, Disc loss: 0.7090]
[2025-03-20 11:20:43,971: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:20:43,974: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:20:43,975: INFO: common: created directory at: artifacts]
[2025-03-20 11:20:43,976: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:20:43,977: INFO: 4176678459: Loading GAN models]
[2025-03-20 11:20:44,560: INFO: 4176678459: Loading test data]
[2025-03-20 11:20:44,730: INFO: 4176678459: Generating Monet-style samples]
[2025-03-20 11:20:48,134: INFO: 4176678459: Calculating FID score]
[2025-03-20 11:21:31,942: INFO: 4176678459: FID Score: 458.2857848227136]
[2025-03-20 11:21:31,972: INFO: 4176678459: Saving generated images]
[2025-03-20 11:21:31,979: INFO: common: created directory at: artifacts/evaluation/generated_samples]
[2025-03-20 11:21:33,718: INFO: common: json file saved at: artifacts/evaluation/metrics.json]
[2025-03-20 11:21:33,724: ERROR: 4176678459: Error during GAN evaluation: isinstance() arg 2 must be a type or tuple of types]
Traceback (most recent call last):
  File "/var/folders/78/nst9ky854lx72gbdr2th76500000gn/T/ipykernel_27631/4176678459.py", line 277, in <module>
    scores = evaluation.evaluate()
  File "/var/folders/78/nst9ky854lx72gbdr2th76500000gn/T/ipykernel_27631/4176678459.py", line 231, in evaluate
    save_json(path=Path(os.path.join(self.config.evaluation_output_path, "metrics.json")), data=scores)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/ensure/main.py", line 850, in __call__
    if not isinstance(return_val, self.return_templ):
TypeError: isinstance() arg 2 must be a type or tuple of types
[2025-03-20 11:23:54,388: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:23:54,391: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:23:54,392: INFO: common: created directory at: artifacts]
[2025-03-20 11:23:54,393: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:23:54,394: INFO: 743645840: Loading GAN models]
[2025-03-20 11:23:54,998: INFO: 743645840: Loading test data]
[2025-03-20 11:23:55,175: INFO: 743645840: Generating Monet-style samples]
[2025-03-20 11:23:58,349: INFO: 743645840: Calculating FID score]
[2025-03-20 11:24:18,979: INFO: 743645840: FID Score: 458.2857848227136]
[2025-03-20 11:24:18,987: INFO: 743645840: Saving generated images]
[2025-03-20 11:24:18,990: INFO: common: created directory at: artifacts/evaluation/generated_samples]
[2025-03-20 11:24:19,646: INFO: common: json file saved at: artifacts/evaluation/metrics.json]
[2025-03-20 11:24:19,647: ERROR: 743645840: Error during GAN evaluation: isinstance() arg 2 must be a type or tuple of types]
Traceback (most recent call last):
  File "/var/folders/78/nst9ky854lx72gbdr2th76500000gn/T/ipykernel_27687/743645840.py", line 291, in <module>
    scores = evaluation.evaluate()
  File "/var/folders/78/nst9ky854lx72gbdr2th76500000gn/T/ipykernel_27687/743645840.py", line 240, in evaluate
    save_json(path=Path(os.path.join(self.config.evaluation_output_path, "metrics.json")), data=scores)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/ensure/main.py", line 850, in __call__
    if not isinstance(return_val, self.return_templ):
TypeError: isinstance() arg 2 must be a type or tuple of types
[2025-03-20 11:25:13,112: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:25:13,115: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:25:13,116: INFO: common: created directory at: artifacts]
[2025-03-20 11:25:13,117: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:25:13,117: INFO: 844790829: Loading GAN models]
[2025-03-20 11:25:13,638: INFO: 844790829: Loading test data]
[2025-03-20 11:25:13,806: INFO: 844790829: Generating Monet-style samples]
[2025-03-20 11:25:17,726: INFO: 844790829: Calculating FID score]
[2025-03-20 11:25:35,845: INFO: 844790829: FID Score: 458.2857848227136]
[2025-03-20 11:25:35,849: INFO: 844790829: Saving generated images]
[2025-03-20 11:25:35,852: INFO: common: created directory at: artifacts/evaluation/generated_samples]
[2025-03-20 11:25:36,528: INFO: 844790829: Metrics saved to artifacts/evaluation/metrics.json]
[2025-03-20 11:25:36,528: INFO: 844790829: Logging to MLflow]
[2025-03-20 11:25:36,677: ERROR: 844790829: Error during GAN evaluation: The configured tracking uri scheme: 'file' is invalid for use with the proxy mlflow-artifact scheme. The allowed tracking schemes are: {'https', 'http'}]
Traceback (most recent call last):
  File "/var/folders/78/nst9ky854lx72gbdr2th76500000gn/T/ipykernel_27715/844790829.py", line 292, in <module>
    evaluation.log_into_mlflow()
  File "/var/folders/78/nst9ky854lx72gbdr2th76500000gn/T/ipykernel_27715/844790829.py", line 267, in log_into_mlflow
    mlflow.log_artifact(self.comparison_path, "generated_samples")
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/tracking/fluent.py", line 783, in log_artifact
    MlflowClient().log_artifact(run_id, local_path, artifact_path)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/tracking/client.py", line 1023, in log_artifact
    self._tracking_client.log_artifact(run_id, local_path, artifact_path)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/tracking/_tracking_service/client.py", line 431, in log_artifact
    artifact_repo = self._get_artifact_repo(run_id)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/tracking/_tracking_service/client.py", line 416, in _get_artifact_repo
    artifact_repo = get_artifact_repository(artifact_uri)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/store/artifact/artifact_repository_registry.py", line 106, in get_artifact_repository
    return _artifact_repository_registry.get_artifact_repository(artifact_uri)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/store/artifact/artifact_repository_registry.py", line 72, in get_artifact_repository
    return repository(artifact_uri)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/store/artifact/mlflow_artifacts_repo.py", line 45, in __init__
    super().__init__(self.resolve_uri(artifact_uri, get_tracking_uri()))
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/store/artifact/mlflow_artifacts_repo.py", line 59, in resolve_uri
    _validate_uri_scheme(track_parse.scheme)
  File "/Applications/AI/GAN/.venv/lib/python3.9/site-packages/mlflow/store/artifact/mlflow_artifacts_repo.py", line 35, in _validate_uri_scheme
    raise MlflowException(
mlflow.exceptions.MlflowException: The configured tracking uri scheme: 'file' is invalid for use with the proxy mlflow-artifact scheme. The allowed tracking schemes are: {'https', 'http'}
[2025-03-20 11:26:49,310: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:26:49,313: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:26:49,314: INFO: common: created directory at: artifacts]
[2025-03-20 11:26:49,314: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:26:49,315: INFO: 4243899433: Loading GAN models]
[2025-03-20 11:26:49,875: INFO: 4243899433: Loading test data]
[2025-03-20 11:26:50,052: INFO: 4243899433: Generating Monet-style samples]
[2025-03-20 11:26:53,774: INFO: 4243899433: Calculating FID score]
[2025-03-20 11:27:14,694: INFO: 4243899433: FID Score: 458.2857848227136]
[2025-03-20 11:27:14,709: INFO: 4243899433: Saving generated images]
[2025-03-20 11:27:14,723: INFO: common: created directory at: artifacts/evaluation/generated_samples]
[2025-03-20 11:27:15,632: INFO: 4243899433: Metrics saved to artifacts/evaluation/metrics.json]
[2025-03-20 11:27:15,632: INFO: 4243899433: Attempting to log to MLflow]
[2025-03-20 11:27:15,633: WARNING: 4243899433: MLflow tracking URI scheme 'file' not supported for artifact logging]
[2025-03-20 11:27:15,633: INFO: 4243899433: To use MLflow, configure a valid tracking server with http/https]
[2025-03-20 11:27:15,634: INFO: 4243899433: Evaluation completed successfully! Metrics: {'fid_score': 458.2857848227136}]
[2025-03-20 11:28:50,714: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:28:50,717: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:28:50,718: INFO: common: created directory at: artifacts]
[2025-03-20 11:28:50,719: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:28:50,719: INFO: 4243899433: Loading GAN models]
[2025-03-20 11:28:51,229: INFO: 4243899433: Loading test data]
[2025-03-20 11:28:51,397: INFO: 4243899433: Generating Monet-style samples]
[2025-03-20 11:28:54,356: INFO: 4243899433: Calculating FID score]
[2025-03-20 11:29:23,987: INFO: 4243899433: FID Score: 458.2857848227136]
[2025-03-20 11:29:23,992: INFO: 4243899433: Saving generated images]
[2025-03-20 11:29:24,002: INFO: common: created directory at: artifacts/evaluation/generated_samples]
[2025-03-20 11:29:24,811: INFO: 4243899433: Metrics saved to artifacts/evaluation/metrics.json]
[2025-03-20 11:29:24,812: INFO: 4243899433: Attempting to log to MLflow]
[2025-03-20 11:29:24,813: WARNING: 4243899433: MLflow tracking URI scheme 'file' not supported for artifact logging]
[2025-03-20 11:29:24,814: INFO: 4243899433: To use MLflow, configure a valid tracking server with http/https]
[2025-03-20 11:29:24,814: INFO: 4243899433: Evaluation completed successfully! Metrics: {'fid_score': 458.2857848227136}]
[2025-03-20 11:34:48,437: INFO: main: >>>>>> stage GAN Evaluation stage started <<<<<<]
[2025-03-20 11:34:48,441: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:34:48,443: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:34:48,443: INFO: common: created directory at: artifacts]
[2025-03-20 11:34:48,443: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:34:48,444: INFO: evaluate_gan: Loading GAN models]
[2025-03-20 11:34:48,872: INFO: evaluate_gan: Loading test data]
[2025-03-20 11:34:48,902: INFO: evaluate_gan: Found images in artifacts/data_ingestion/gan-getting-started/photo_jpg and artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 11:34:49,044: INFO: evaluate_gan: Generating Monet-style samples]
[2025-03-20 11:34:53,149: ERROR: evaluate_gan: Error calculating metrics: Input 0 of layer "model_1" is incompatible with the layer: expected shape=(None, 256, 256, 3), found shape=(32, 128, 128, 3)]
[2025-03-20 11:34:53,150: INFO: evaluate_gan: Saving generated images]
[2025-03-20 11:34:54,354: INFO: evaluate_gan: Attempting to log to MLflow]
[2025-03-20 11:34:54,354: WARNING: evaluate_gan: MLflow tracking URI scheme 'file' not supported]
[2025-03-20 11:34:54,355: INFO: main: >>>>>> stage GAN Evaluation stage completed <<<<<<

x==========x]
[2025-03-20 11:34:54,355: INFO: main: Evaluation metrics: {}]
[2025-03-20 11:36:41,831: INFO: main: >>>>>> stage GAN Evaluation stage started <<<<<<]
[2025-03-20 11:36:41,834: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:36:41,836: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:36:41,836: INFO: common: created directory at: artifacts]
[2025-03-20 11:36:41,836: INFO: common: created directory at: artifacts/evaluation]
[2025-03-20 11:36:41,836: INFO: evaluate_gan: Loading GAN models]
[2025-03-20 11:36:42,226: INFO: evaluate_gan: Loading test data]
[2025-03-20 11:36:42,256: INFO: evaluate_gan: Found images in artifacts/data_ingestion/gan-getting-started/photo_jpg and artifacts/data_ingestion/gan-getting-started/monet_jpg]
[2025-03-20 11:36:42,382: INFO: evaluate_gan: Generating Monet-style samples]
[2025-03-20 11:36:45,524: INFO: evaluate_gan: Resizing generated images from (128, 128) to (256, 256)]
[2025-03-20 11:36:48,766: INFO: evaluate_gan: Real score: -0.0906, Fake score: -0.2719]
[2025-03-20 11:36:48,768: INFO: evaluate_gan: Saving generated images]
[2025-03-20 11:36:50,418: INFO: evaluate_gan: Attempting to log to MLflow]
[2025-03-20 11:36:50,418: WARNING: evaluate_gan: MLflow tracking URI scheme 'file' not supported]
[2025-03-20 11:36:50,421: INFO: main: >>>>>> stage GAN Evaluation stage completed <<<<<<

x==========x]
[2025-03-20 11:36:50,421: INFO: main: Evaluation metrics: {'avg_real_score': -0.09057275950908661, 'avg_fake_score': -0.271882563829422, 'score_difference': 0.1813098043203354}]
[2025-03-20 11:54:35,488: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:54:35,491: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:54:35,493: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:54:35,494: INFO: common: created directory at: artifacts]
[2025-03-20 11:54:35,496: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:54:35,841: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:54:35,889: INFO: _internal: [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8080
 * Running on http://172.16.29.87:8080]
[2025-03-20 11:54:35,890: INFO: _internal: [33mPress CTRL+C to quit[0m]
[2025-03-20 11:54:35,891: INFO: _internal:  * Restarting with stat]
[2025-03-20 11:54:39,888: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:54:39,891: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:54:39,893: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:54:39,893: INFO: common: created directory at: artifacts]
[2025-03-20 11:54:39,894: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:54:40,193: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:54:40,219: WARNING: _internal:  * Debugger is active!]
[2025-03-20 11:54:40,244: INFO: _internal:  * Debugger PIN: 216-093-100]
[2025-03-20 11:54:52,253: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:54:52] "[35m[1mGET / HTTP/1.1[0m" 500 -]
[2025-03-20 11:54:52,300: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:54:52] "GET /?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1" 200 -]
[2025-03-20 11:54:52,302: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:54:52] "GET /?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1" 200 -]
[2025-03-20 11:54:52,319: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:54:52] "GET /?__debugger__=yes&cmd=resource&f=console.png&s=pJpf6fRPQJ2lxqsHIWU3 HTTP/1.1" 200 -]
[2025-03-20 11:54:52,969: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:54:52] "GET /?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1" 200 -]
[2025-03-20 11:57:01,399: INFO: _internal:  * Detected change in '/Applications/AI/GAN/web_app.py', reloading]
[2025-03-20 11:57:01,915: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 11:57:11,023: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:57:11,027: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:57:11,029: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:57:11,029: INFO: common: created directory at: artifacts]
[2025-03-20 11:57:11,030: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:57:11,402: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:57:11,440: INFO: _internal: [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8080
 * Running on http://172.16.29.87:8080]
[2025-03-20 11:57:11,440: INFO: _internal: [33mPress CTRL+C to quit[0m]
[2025-03-20 11:57:11,441: INFO: _internal:  * Restarting with stat]
[2025-03-20 11:57:15,296: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:57:15,299: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:57:15,302: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:57:15,302: INFO: common: created directory at: artifacts]
[2025-03-20 11:57:15,303: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:57:15,589: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:57:15,609: WARNING: _internal:  * Debugger is active!]
[2025-03-20 11:57:15,624: INFO: _internal:  * Debugger PIN: 216-093-100]
[2025-03-20 11:57:35,708: INFO: _internal:  * Detected change in '/Applications/AI/GAN/web_app.py', reloading]
[2025-03-20 11:57:36,175: INFO: _internal:  * Restarting with stat]
[2025-03-20 11:57:36,197: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 11:57:44,518: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:57:44,521: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:57:44,523: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:57:44,523: INFO: common: created directory at: artifacts]
[2025-03-20 11:57:44,525: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:57:44,846: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:57:44,883: INFO: _internal: [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8080
 * Running on http://172.16.29.87:8080]
[2025-03-20 11:57:44,883: INFO: _internal: [33mPress CTRL+C to quit[0m]
[2025-03-20 11:57:44,884: INFO: _internal:  * Restarting with stat]
[2025-03-20 11:57:48,857: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:57:48,860: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:57:48,862: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:57:48,863: INFO: common: created directory at: artifacts]
[2025-03-20 11:57:48,864: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:57:49,147: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:57:49,172: WARNING: _internal:  * Debugger is active!]
[2025-03-20 11:57:49,185: INFO: _internal:  * Debugger PIN: 216-093-100]
[2025-03-20 11:57:58,334: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:57:58] "[35m[1mGET / HTTP/1.1[0m" 500 -]
[2025-03-20 11:57:58,387: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:57:58] "[36mGET /?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1[0m" 304 -]
[2025-03-20 11:57:58,387: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:57:58] "[36mGET /?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1[0m" 304 -]
[2025-03-20 11:57:58,401: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:57:58] "GET /?__debugger__=yes&cmd=resource&f=console.png&s=3I4VrZbXK7khO46LCbJf HTTP/1.1" 200 -]
[2025-03-20 11:59:16,365: INFO: _internal:  * Detected change in '/Applications/AI/GAN/web_app.py', reloading]
[2025-03-20 11:59:16,894: INFO: _internal:  * Restarting with stat]
[2025-03-20 11:59:23,359: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:59:23,362: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:59:23,364: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:59:23,364: INFO: common: created directory at: artifacts]
[2025-03-20 11:59:23,366: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:59:23,716: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:59:23,747: WARNING: _internal:  * Debugger is active!]
[2025-03-20 11:59:23,761: INFO: _internal:  * Debugger PIN: 216-093-100]
[2025-03-20 11:59:25,759: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 11:59:26,013: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 11:59:36,228: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:59:36,232: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:59:36,234: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:59:36,234: INFO: common: created directory at: artifacts]
[2025-03-20 11:59:36,235: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:59:36,581: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:59:36,622: INFO: _internal: [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8080
 * Running on http://172.16.29.87:8080]
[2025-03-20 11:59:36,622: INFO: _internal: [33mPress CTRL+C to quit[0m]
[2025-03-20 11:59:36,623: INFO: _internal:  * Restarting with stat]
[2025-03-20 11:59:40,946: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 11:59:40,949: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 11:59:40,952: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 11:59:40,952: INFO: common: created directory at: artifacts]
[2025-03-20 11:59:40,954: INFO: web_app: Loading generator model for web app]
[2025-03-20 11:59:41,267: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 11:59:41,286: WARNING: _internal:  * Debugger is active!]
[2025-03-20 11:59:41,300: INFO: _internal:  * Debugger PIN: 216-093-100]
[2025-03-20 11:59:41,436: INFO: _internal: 172.16.29.87 - - [20/Mar/2025 11:59:41] "[35m[1mGET / HTTP/1.1[0m" 500 -]
[2025-03-20 11:59:41,437: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:59:41] "[35m[1mGET / HTTP/1.1[0m" 500 -]
[2025-03-20 11:59:41,471: INFO: _internal: 172.16.29.87 - - [20/Mar/2025 11:59:41] "GET /?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1" 200 -]
[2025-03-20 11:59:41,471: INFO: _internal: 172.16.29.87 - - [20/Mar/2025 11:59:41] "GET /?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1" 200 -]
[2025-03-20 11:59:41,486: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:59:41] "[36mGET /?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1[0m" 304 -]
[2025-03-20 11:59:41,489: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:59:41] "[36mGET /?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1[0m" 304 -]
[2025-03-20 11:59:41,650: INFO: _internal: 172.16.29.87 - - [20/Mar/2025 11:59:41] "GET /?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1" 200 -]
[2025-03-20 11:59:41,652: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 11:59:41] "GET /?__debugger__=yes&cmd=resource&f=console.png&s=03hQSIeeI3vy7xrWUEmZ HTTP/1.1" 200 -]
[2025-03-20 12:00:40,701: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 12:00:40,934: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 12:01:50,411: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 12:01:50,414: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 12:01:50,416: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 12:01:50,417: INFO: common: created directory at: artifacts]
[2025-03-20 12:01:50,417: INFO: web_app: Template folder: /Applications/AI/GAN/templates]
[2025-03-20 12:01:50,417: INFO: web_app: Static folder: /Applications/AI/GAN/static]
[2025-03-20 12:01:50,418: ERROR: main: 'GANWebApp' object has no attribute 'setup_routes']
Traceback (most recent call last):
  File "/Applications/AI/GAN/main.py", line 13, in <module>
    web_app.main()
  File "/Applications/AI/GAN/src/GAN/pipeline/stage_05_web_app.py", line 15, in main
    web_app = GANWebApp(config=web_app_config)
  File "/Applications/AI/GAN/src/GAN/components/web_app.py", line 39, in __init__
    self.setup_routes()
AttributeError: 'GANWebApp' object has no attribute 'setup_routes'
[2025-03-20 12:03:00,990: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 12:03:00,994: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 12:03:00,996: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 12:03:00,996: INFO: common: created directory at: artifacts]
[2025-03-20 12:03:00,996: INFO: web_app: Template folder: /Applications/AI/GAN/templates]
[2025-03-20 12:03:00,996: INFO: web_app: Static folder: /Applications/AI/GAN/static]
[2025-03-20 12:03:00,997: INFO: web_app: Loading generator model for web app]
[2025-03-20 12:03:01,354: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 12:03:01,401: INFO: _internal: [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8080
 * Running on http://172.16.29.87:8080]
[2025-03-20 12:03:01,401: INFO: _internal: [33mPress CTRL+C to quit[0m]
[2025-03-20 12:03:01,402: INFO: _internal:  * Restarting with stat]
[2025-03-20 12:03:05,828: INFO: main: >>>>>> stage Web Application Deployment stage started <<<<<<]
[2025-03-20 12:03:05,831: INFO: common: yaml file: config/config.yaml loaded successfully]
[2025-03-20 12:03:05,833: INFO: common: yaml file: params.yaml loaded successfully]
[2025-03-20 12:03:05,834: INFO: common: created directory at: artifacts]
[2025-03-20 12:03:05,834: INFO: web_app: Template folder: /Applications/AI/GAN/templates]
[2025-03-20 12:03:05,834: INFO: web_app: Static folder: /Applications/AI/GAN/static]
[2025-03-20 12:03:05,835: INFO: web_app: Loading generator model for web app]
[2025-03-20 12:03:06,147: INFO: web_app: Starting web server on 0.0.0.0:8080]
[2025-03-20 12:03:06,171: WARNING: _internal:  * Debugger is active!]
[2025-03-20 12:03:06,191: INFO: _internal:  * Debugger PIN: 216-093-100]
[2025-03-20 12:03:06,247: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 12:03:06] "GET / HTTP/1.1" 200 -]
[2025-03-20 12:03:06,287: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 12:03:06] "GET /static/js/script.js HTTP/1.1" 200 -]
[2025-03-20 12:03:06,287: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 12:03:06] "GET /static/css/style.css HTTP/1.1" 200 -]
[2025-03-20 12:03:06,494: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 12:03:06] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -]
[2025-03-20 12:03:29,711: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 12:03:29] "POST /generate HTTP/1.1" 200 -]
[2025-03-20 12:03:29,743: INFO: _internal: 127.0.0.1 - - [20/Mar/2025 12:03:29] "[36mGET /static/css/style.css HTTP/1.1[0m" 304 -]
[2025-03-20 12:03:45,800: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
[2025-03-20 12:03:46,058: INFO: main: >>>>>> stage Web Application Deployment stage completed <<<<<<

x==========x]
